app/controllers/v1/songs_controller.rb

```diff
@@ -1,11 +1,11 @@
 class V1::SongsController < ApplicationController
   def index
-    songs = Song.includes(:artist)
+    songs = Song.includes(:artist).decorate
     render json: songs, include: [:artist]
   end
 
   def show
-    song = Song.find(params[:id])
+    song = Song.find(params[:id]).decorate
```

Comment:
So, this is something not 100% obvious, but since decorating the model "wraps" it, it's not the best practice to pass around decorated models from the controller into the view -- and essentially the serializer/json here is acting as the view.

The author of Draper says why here: https://github.com/drapergem/draper#when-to-decorate-objects. 

To avoid having the pass around the decorated model, he's made a helper shown in this example as `decorates_assigned`:

```
# app/controllers/articles_controller.rb
class ArticlesController < ApplicationController
  decorates_assigned :article

  def show
    @article = Article.find(params[:id])
  end
end
```

The decorates_assigned :article bit is roughly equivalent to

```
def article
  @decorated_article ||= @article.decorate
end
helper_method :article
```

But, since we're not really passing into a view or .erb, it can be better to but that method in the serializer itself.

```
def decorated_song
  @decorated_song ||= @song.decorate
end
```

And then only in the serializer methods that need something decorate, use

```
def full_title
  decorated_song.full_title
end
```